### ISSUE TITLE
Add `GODOT_PROFILING_FUNCTION` support in GDExtension (similar to GDNative in 3.x)

### ISSUE THREAD

**Issue by Withaust** [godotengine/godot-cpp#1032]
Previously in `GDNative` there was a handy `GODOT_PROFILING_FUNCTION` macro, which allowed to profile custom C++ methods/functions in the main editor profile, but now in `GDExtention` there does not seem to be a way to achieve the similar result.

**Comment by dalygbarron** [comment]
I see that this is closed as completed, but it is not linked with any code changes to look at and also GODOT_PROFILING_FUNCTION has not been added back to the codebase. If it's ok could I ask how this is going to work from now on if that macro is not being used?

**Comment by Calinou** [comment]
@Withaust Did you find a solution to this problem? If so, please document it here by commenting :slightly_smiling_face:

**Comment by dalygbarron** [comment]
yeah I started looking last night to see if I could sort it out myself and I am pretty sure the functionality no longer exists since it was relying on a call to some kind of gdnative api. If somebody were to point out how similar functionality could be acheived in the current codebase I'd be happy to make a PR, but as it stands I do not know how.

**Comment by Withaust** [comment]
I don't remember why I closed this, but this functionality is still missing.
